packages/unlock-js/src/web3Service.ts

```diff
@@ -609,4 +609,83 @@ export default class Web3Service extends UnlockService {
     )
     return price
   }
+
+  /**
+   * Get uniswap price in the out token.
+   * ```ts
+   * const web3Service = new Web3Service(networks)
+   * const price = await web3Service.consultUniswap({
+   *  network: 1,
+   *  tokenInAddress: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+   *  tokenOutAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
+   *  value: '1',
+   * })
+   *
+   * console.log(price)
+   * ```
+   */
+  async consultUniswap(options: {
+    tokenInAddress: string
+    tokenOutAddress?: string
+    value: string
+    network?: number
+  }) {
+    const { network, tokenInAddress, value } = options
+    const networkId = network || 1
+    const networkConfig = this.networks[networkId] // By default, use mainnet
+    const tokenOutAddress =
+      options.tokenOutAddress ||
```

Comment:
I like that, but lets' add also a `tokenInAddress` to default to the local chain's "wrapped" token, like Eth on mainnet, MATIC if the the network is polygon, BNB if the network is BSC... etc.
That means you need to add the "mainnet" ERC20 version of each chain's token but I think that's doable?

wdyt?

Reply:
Please don't forget!